src/chain.jl

```diff
@@ -0,0 +1,83 @@
+const Transform = Union{Assertion, Filter, Imputor}
+
+"""
+    Chain
+
+Runs multiple `Assertions`, `Filter` or `Imputor`s on the same data in the order they're
+provided.
+
+# Fields
+* `transforms::Vector{Union{Assertion, Filter, Imputor}}`
+"""
+struct Chain
+    transforms::Vector{Transform}
+end
+
+"""
+    Chain(transforms::Union{Assertion, Filter, Imputor}...) -> Chain
+
+Creates a Chain using the transforms provided (ordering matters).
+"""
+Chain(transforms::Transform...) = Chain(collect(transforms))
+
+"""
+Compose new chains with the composition operator
+
+# Example
+
+```jldoctest
+julia> using Impute: Impute, Interpolate, NOCB, LOCF
+
+julia> M = [missing 2.0 missing missing 5.0; 1.1 2.2 missing 4.4 missing]
+2×5 Array{Union{Missing, Float64},2}:
+  missing  2.0  missing   missing  5.0
+ 1.1       2.2  missing  4.4        missing
+
+julia> Impute.run(M, Interpolate() ∘ NOCB() ∘ LOCF(); dims=:rows)
+2×5 Array{Union{Missing, Float64},2}:
+ 2.0  2.0  3.0  4.0  5.0
+ 1.1  2.2  3.3  4.4  4.4
+```
+"""
+Base.:(∘)(a::Transform, b::Transform) = Chain(Transform[a, b])
+function Base.:(∘)(C::Chain, b::Transform)
+    push!(C.transforms, b)
+    return C
+end
```

Comment:
This mutates `C`.
I would find that behavour *very* surprising.

```suggestion
Base.:(∘)(C::Chain, b::Transform) = Chain(c.transforms..., b)
```